lib/cli/parse-args.js

```diff
@@ -0,0 +1,126 @@
+// TODO: Consider publishing as independent package
+
+'use strict';
+
+const ServerlessError = require('../serverless-error');
+
+const paramRe = /^-(?:-(?<fullName>[a-z][A-Za-z0-9:-]+)|(?<aliasNames>[a-z]+))(?:=(?<value>.+)|$)/;
+const isParamName = RegExp.prototype.test.bind(paramRe);
+
+module.exports = (
+  args,
+  { boolean = new Set(), alias = new Map(), string = new Set(), multiple = new Set() }
+) => {
+  const result = Object.create(null);
+  result._ = [];
+  for (let i = 0, arg; (arg = args[i]); ++i) {
+    if (arg === '--') {
+      result._.push(...args.slice(i + 1));
+      break;
+    }
+    const paramMatch = arg.match(paramRe);
+    if (!paramMatch) {
+      result._.push(arg);
+      continue;
+    }
+    const { fullName, aliasNames, value } = paramMatch.groups;
+    const isBoolean = !value && (!args[i + 1] || args[i + 1] === '--' || isParamName(args[i + 1]));
+
+    let paramName;
+    if (aliasNames) {
+      if (aliasNames.length > 1) {
+        if (value) {
+          throw new ServerlessError(
+            `Unexpected value for "--${aliasNames}"`,
+            'UNEXPECTED_CLI_PARAM_VALUE'
+          );
+        }
+        for (const aliasName of aliasNames) {
+          paramName = alias.get(aliasName) || aliasName;
+          if (result[paramName] != null) {
+            throw new ServerlessError(
+              `Unexpected multiple "--${paramName}`,
```

Comment:
this also might be a bit misleading when someone specifies something twice with aliases, but I'm not sure how we could improve that here (if I'm not mistaken of course)

Reply:
Good point, I've updated the message to also contain an alias name in such case

Reply:
awesome :raised_hands: